go/porcelain/deploy.go

```diff
@@ -384,6 +394,57 @@ func (n *Netlify) WaitUntilDeployLive(ctx context.Context, d *models.Deploy) (*m
 	return n.waitForState(ctx, d, "ready")
 }
 
+func (n *Netlify) uploadFilesWithSkipSameFileOptimization(ctx context.Context, d *models.Deploy, files *deployFiles, observer DeployObserver, t uploadType, timeout time.Duration) error {
+	sharedErr := &uploadError{err: nil, mutex: &sync.Mutex{}}
+	sem := make(chan int, n.uploadLimit)
+	wg := &sync.WaitGroup{}
+	defer wg.Wait()
+
+	var required []string
+	switch t {
+	case fileUpload:
+		required = d.Required
+	case functionUpload:
+		required = d.RequiredFunctions
+	}
+
+	count := 0
+	for _, sha := range required {
+		if files, exist := files.Hashed[sha]; exist {
+			for range files {
+				count++
+			}
+		}
+	}
+
+	log := context.GetLogger(ctx)
+	log.Infof("Uploading %v files", count)
+
+	for _, sha := range required {
+		if files, exist := files.Hashed[sha]; exist {
+			file := files[0]
+
+			select {
+			case sem <- 1:
+				wg.Add(1)
+				go n.uploadFile(ctx, d, file, observer, t, timeout, wg, sem, sharedErr)
+			case <-ctx.Done():
+				log.Info("Context terminated, aborting file upload")
+				return errors.Wrap(ctx.Err(), "aborted file upload early")
+			}
+
+			if len(files) > 1 {
```

Comment:
Is this logged to the customer? If it is, I wonder how useful it is to the customer to know we skipped some files during upload.

Reply:
No, all the logs in the Go client use the sys logger:
- https://github.com/netlify/buildbot/blob/c87074660cd9a4568c0ac759bc083987e94204b3/api/netlify_api_client.go#L111